### PR TITLE
Add `get_global()` to `cupy.RawModule`

### LINKED ISSUE #1703: CUDA constant memory management
Hi,

I haven't found any API concerning the creation and management of constant memory. Are you going to release it soon?

In my opinion this feature could be extremely useful due to the introduction of RawKernel API (v5.0.0rc1).

Thanks for your support

### PR DESCRIPTION
**Pull request by leofang**
Close #1703.

This PR adds a public API to `cupy.RawModule`. This enables, for example, the usage of constant memory in `RawKernel`s, which makes it easier for PyCUDA and general CUDA users to migrate to CuPy, part of an ongoing effort tracked in #2450. See also the discussion in #1703.

Note that a Cython wrapper for the CUDA driver API `cuModuleGetGlobal()` already exists, so with this PR all `cuModuleGet*()` functions are exposed to end users via `cupy.RawModule`.

### CODE REVIEW

File: docs/source/tutorial/kernel.rst [hunk 1]
```diff
@@ -278,6 +278,8 @@ For dealing a large raw CUDA source or loading an existing CUDA binary, the :cla
     >>> ker_times((N,), (N,), (x1, x2, y, N**2)) # y = x1 * x2
     >>> assert cp.allclose(y, x1 * x2)
 
+For CUDA kernels that need to access globol symbols, such as constant memory, the :meth:`~cupy.RawModule.get_global` method can be used, see its documentation for further detail.
```
Comment (grlee77) [inline docs/source/tutorial/kernel.rst]: typo: globol -> global

Comment (leofang) [reply docs/source/tutorial/kernel.rst]: Thanks, @grlee77! Good catch.

**Comment by kmaehashi** [comment]
@leofang Could you resolve conflicts? Thanks.

**Comment by leofang** [comment]
@kmaehashi Thanks for the heads-up. I just resolved it.

File: tests/cupy_tests/core_tests/test_raw.py [hunk 2]
```diff
@@ -367,6 +382,17 @@ def test_get_function_failure(self):
             self.mod2.get_function('no_such_kernel')
         assert 'CUDA_ERROR_NOT_FOUND' in str(ex.value)
 
+    def test_const_memory(self):
+        mod = cupy.RawModule(test_const_mem, backend=self.backend)
+        ker = mod.get_function('multiply_by_const')
+        mem_ptr = mod.get_global('some_array')
+        const_arr = cupy.ndarray((100,), cupy.float32, mem_ptr)
+        const_data = cupy.arange(100, dtype=cupy.float32)
```
Comment (leofang) [inline tests/cupy_tests/core_tests/test_raw.py]:
```suggestion
        data = cupy.arange(100, dtype=cupy.float32)
```

File: tests/cupy_tests/core_tests/test_raw.py [hunk 3]
```diff
@@ -367,6 +382,17 @@ def test_get_function_failure(self):
...
+        const_arr[...] = const_data
```
Comment (leofang) [inline tests/cupy_tests/core_tests/test_raw.py]:
```suggestion
        const_arr[...] = data
```

File: tests/cupy_tests/core_tests/test_raw.py [hunk 4]
```diff
@@ -367,6 +382,17 @@ def test_get_function_failure(self):
...
+        output_arr = cupy.ones(100, dtype=cupy.float32)
+        ker((1,), (100,), (output_arr, cupy.int32(100)))
+        assert (const_data == output_arr).all()
```
Comment (leofang) [inline tests/cupy_tests/core_tests/test_raw.py]:
```suggestion
        assert (data == output_arr).all()
```

**Comment by leofang** [comment]
@kmaehashi PTAL. Thanks.

**Comment by leofang** [comment]
@kmaehashi Can this PR be added to v8.0.0a1? Thanks.

File: tests/cupy_tests/core_tests/test_raw.py [hunk 5]
```diff
@@ -367,6 +382,17 @@ def test_get_function_failure(self):
...
 
+    def test_const_memory(self):
+        mod = cupy.RawModule(test_const_mem, backend=self.backend)
+        ker = mod.get_function('multiply_by_const')
+        mem_ptr = mod.get_global('some_array')
+        const_arr = cupy.ndarray((100,), cupy.float32, mem_ptr)
+        data = cupy.arange(100, dtype=cupy.float32)
+        const_arr[...] = data
+        output_arr = cupy.ones(100, dtype=cupy.float32)
+        ker((1,), (100,), (output_arr, cupy.int32(100)))
```
Comment (emcastillo) [inline tests/cupy_tests/core_tests/test_raw.py]:
```suggestion
        ker((1,), (100,), (output_arr, 100))
```

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
@emcastillo Thanks. Apparently this branch is too outdated...I just rebased and force-pushed. PTAL.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
Something is wrong with Jenkins?

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**